pycbc/population/live_pastro_utils.py

```diff
@@ -71,6 +76,24 @@ def __init__(self, specfile, bank):
             self.spec = _check_spec[self.method](self.spec_json)
             self.bank = _read_bank[self.method](self.spec, bank)
 
+    def apply_significance_limits(self, trigger_data):
+        """
+        If the network SNR and FAR indicate saturation of the FAR estimate,
+        set them to the fixed values given in the specification.
+        """
+        # This only happens for double or triple events
+        if len(trigger_data['triggered']) > 1:
+            farlim = self.spec['limit_far']
+            snrlim = self.spec['limit_snr']
+            if trigger_data['far'] < farlim and \
+                    trigger_data['network_snr'] > snrlim:
+                logging.debug('Truncating FAR and SNR from %f, %f to %f, %f',
+                              trigger_data['far'], trigger_data['network_snr'],
+                              farlim, snrlim)
+                trigger_data['network_snr'] = snrlim
+                trigger_data['far'] = farlim
```

Comment:
My usual suggestion of negating these if statements, so as to return as early as possible and reduce indentation and nesting.